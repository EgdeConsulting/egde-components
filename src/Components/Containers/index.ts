export * from './PageContainer';
export * from './TitleContainer';

