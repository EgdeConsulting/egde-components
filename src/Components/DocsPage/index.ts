export * from './DocsPageHeader';
export * from './DocsPageBody';
