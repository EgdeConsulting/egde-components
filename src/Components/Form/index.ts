export * from './BaseInput';
export * from './FileInput';
export * from './NumberInput';
export * from './TextAreaInput';
export * from './TextInput';
export * from './Checkbox';
export * from './Radio';
export * from './Button';
export * from './Select';
