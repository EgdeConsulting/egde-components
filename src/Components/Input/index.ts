export * from './BaseInput';
export * from './FileInput';
export * from './NumberInput';
export * from './TextAreaInput';
export * from './TextInput';
