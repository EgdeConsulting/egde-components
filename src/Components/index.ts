export * from './Alert';
export * from './Button';
export * from './Card';
export * from './Checkbox';
export * from './DeleteModal';
export * from './Input';
export * from './Modal';
export * from './PageContainer';
export * from './Radio';
export * from './Tabs';

