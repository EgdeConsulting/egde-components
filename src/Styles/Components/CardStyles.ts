const CardStyles = {
    baseStyle: {
        backgroundColor: 'lightblue',
        borderRadius: '6px',
        width: '25rem',
        height: '11rem',
        px: '1.5rem',
        paddingTop: '1rem',
        paddingBottom: '0',
        _hover: {
            outline: '2px solid',
            outlineColor: 'blue',
            cursor: 'pointer',
        },
    },
};

export { CardStyles };
