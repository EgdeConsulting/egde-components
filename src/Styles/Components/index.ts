export * from './Text';
export * from './Heading';
export * from './Button';
export * from './IconButton';
export * from './Card';
export * from './Tabs';
export * from './Checkbox';
export * from './Radio';
export * from './Steps';
export * from './Input';
export * from './Textarea';
export * from './Table';
export * from './FileInput';
/* export * from './DatePickerStyles';
 */
