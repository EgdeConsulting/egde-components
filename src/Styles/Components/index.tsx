export * from './TextStyles';
export * from './HeadingStyles';
export * from './ButtonStyles';
export * from './CardStyles';
export * from './TabStyles';
export * from './CheckboxStyles';
export * from './RadioStyles';
export * from './StepsStyles';
export * from './InputStyles';
export * from './TextareaStyles';
export * from './TableStyles';
export * from './FileInputStyles';
/* export * from './DatePickerStyles';
 */
