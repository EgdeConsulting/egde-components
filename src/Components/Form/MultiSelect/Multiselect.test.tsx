import '@testing-library/jest-dom';
import { render} from '@testing-library/react';
import React, { useState } from 'react';
import { MultiSelect } from 'Components';
import { SelectOption } from 'Types';

const MULTISELECT_PLACEHOLDER = 'DatePicker placeholder';
const MULTISELECT_LABEL = 'RichTextAreaInput label';

const MultiSelectTester = (): JSX.Element => {
    const [value, setValue] = useState<SelectOption[]>([]);

    return (
        <>
            <MultiSelect
                value={value}
                onChange={setValue}
                placeholder={MULTISELECT_PLACEHOLDER}
                label={MULTISELECT_LABEL}
                options={[
                    { label: 'Option 1', value: 1 },
                    { label: 'Option 2', value: 2},
                    { label: 'Option 3', value: 3 },
                ]}
            />
        </>
    );
}


describe('MultiSelect component test', () => {
    it('should render correctly', () => {
        const tree = render(<MultiSelectTester/>);
        expect(tree).toMatchSnapshot();
    });
});
