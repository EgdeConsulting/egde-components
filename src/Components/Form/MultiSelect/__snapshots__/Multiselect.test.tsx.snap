// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect component test matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-form-control css-oo2x8u"
        role="group"
      >
        <label
          class="chakra-form__label css-60lnuh"
          for="field-1"
          id="field-1-label"
        >
          RichTextAreaInput label
        </label>
        <div
          class="css-0"
        >
          <div
            role="none"
          >
            <div
              class=" css-tlt8t9"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-2-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class=" css-p9orrc"
              >
                <div
                  class=" css-o38mw0"
                >
                  <div
                    class=" css-179kski"
                    id="react-select-2-placeholder"
                  >
                    DatePicker placeholder
                  </div>
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-2-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-readonly="true"
                    class="css-mohuvp-dummyInput-DummyInput"
                    id="field-1"
                    inputmode="none"
                    role="combobox"
                    tabindex="0"
                    value=""
                  />
                </div>
                <div
                  class=" css-hfbj6y"
                >
                  <div
                    aria-hidden="true"
                    class=" css-4zi1wo"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-form-control css-oo2x8u"
      role="group"
    >
      <label
        class="chakra-form__label css-60lnuh"
        for="field-1"
        id="field-1-label"
      >
        RichTextAreaInput label
      </label>
      <div
        class="css-0"
      >
        <div
          role="none"
        >
          <div
            class=" css-tlt8t9"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-p9orrc"
            >
              <div
                class=" css-o38mw0"
              >
                <div
                  class=" css-179kski"
                  id="react-select-2-placeholder"
                >
                  DatePicker placeholder
                </div>
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-readonly="true"
                  class="css-mohuvp-dummyInput-DummyInput"
                  id="field-1"
                  inputmode="none"
                  role="combobox"
                  tabindex="0"
                  value=""
                />
              </div>
              <div
                class=" css-hfbj6y"
              >
                <div
                  aria-hidden="true"
                  class=" css-4zi1wo"
                >
                  <svg
                    class="chakra-icon css-onkibi"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
